submariner/centralized-control-plane.md

```diff
@@ -0,0 +1,386 @@
+  and namespace as labels or annotations on the `ServiceImport` object. Alternately, we could
+  extend the objects' `Spec` to explicitly support a different target service name and namespace.
+  Backward compatibility is provided by defaulting to the object namespace and name if the
+  `Spec` does not define a different destination.
+1. Selective imports (i.e., import services into a subset of clusters in the ClusterSet).
+  This can be enabled as an overlay management plane on top of MCS and allows for Services
+  to be imported into specific clusters only. Since the MCS specification does not dictate
+  how `ServiceImport` objects are communicated in a cluster set, this could be done by the
+  controller without changes to MCS objects.
+1. Centralized control over exporting and importing (i.e., defined in Broker, not
+  workload clusters). This too can be an overlay atop MCS with a management plane creating
+  the corresponding `ServiceExport` and `ServiceImport` objects in the respective clusters.
+
+The first aspect (name independence) makes `ServiceExport` and `ServiceImport` less useful
+ as the abstraction in a central shared location. For example, since all imports are created
+ in the Submariner broker namespace of the ClusterSet, Submariner resorts to tagging the
+ destination namespace and name on the `ServiceImport` using labels and annotations. It might
+ have been clearer to explicitly set those in `ServiceImport.Spec`, if MCS provided that.
+ Similarly, the assumption on name sameness has implication on `ServiceExport` (e.g., two
+ completely independent cluster, owned and managed by different teams, must agree on using
+ unique namespaces and/or names, lest their exports conflate when shared).
+
+It should be further noted that the MCS API specification (intentionally?) leaves the following
+ aspects out of scope, and as a responsibility of some other controller components.
+
+- The mechanism to propagate Exports and Import from/to workload clusters.
+  A synchronization/distribution mechanism is required (e.g., Submariner's Lighthouse is
+  one such example). The mechanism is implementation specific; and
+- Policy objects (e.g., how to declaratively define which exports to import where and with
+  what local workload clusters attributes). These are similarly implementation specific.
+
+While all three (naming, synchronization and policies) could be considered as possible
+ enhancements to the MCS API, it would be worthwhile to experiment with different approaches,
+ such as the one defined here, before attempting to reach community consensus on MCS API changes.
+
+The design proposal attempts to achieve the above goals minimal changes to workload
+ clusters, especially with respect to data plane configuration. Ideally, the control
+ plane changes proposed will are extensions to MCS or implemented as a layer above
+ MCS and confined to the Broker cluster only.
```

Comment:
It is not intended to deviate from MCS API in Lighthouse, but if you implement your own service discovery layer you can define your own behaviors that will fit your needs. This shouldn't be part of submariner though.